### DISCUSSION TITLE
M600 Filament Change Issue

### DISCUSSION THREAD

**Discussion by embraceentropy** [Jyers/Marlin#714] [Q&A] [answered]
I'm using pre-compiled 1.3.2 with 5x5 manual mesh. Trying to use the filament change script in Cura, which just puts M600 at a specified layer. I have my filament load/unload lengths set in EEPROM (both set at 420 ((nice)) and have tested it by issuing M600 direct to the printer with no issues. 

During my print I hit the layer, the nozzle parks, filament change goes fine. However, just before the nozzle returns to the part the extruder backs the filament out maybe ~20-50mm (seems to vary based on the GCODE file). Then the nozzle continues to draw the layer and move the extruder forward/retract/forward, etc as normal, but the filament is backed out of the hotend so nothing is coming out of the nozzle. 

Looking at the code everything seems normal. I'm wondering if the G1 after the M600 setting the extruder absolute position is causing the issue. 

My code is as follows in that area:

G1 X114.894 Y100.914 E596.95416
G0 F9000 X114.93 Y100.83
;MESH:NONMESH
G0 F300 X114.93 Y100.83 Z26
G0 F9000 X114.623 Y100.995
;TIME_ELAPSED:988.854497
M600 X0.00 Y0.00 ; Generated by FilamentChange plugin
;LAYER:86
;TYPE:WALL-INNER
;MESH:B STAMP.STL
G1 F600 X114.409 Y101.068 E596.96544
G1 X114.383 Y101.077 E596.96682
G1 X114.378 Y101.019
G1 X114.372 Y100.962 E596.96833

**Comment by xantrk** [comment]
https://github.com/Jyers/Marlin/discussions/597#discussion-3336795
Similar issue here

**Comment by vhristenco** [comment]
I don't know what configs have the precompiled firmwares, but you can find the M600 settings under config_adv.h

**Comment by Cal-LF** [comment, marked as answer]
I posted #597 comment with the same issue. only when compiling my own. The precompiles work as normal. The precompiles are derived from the same templates that are posted for us to used to compile. Doesn't make sense this should happen.

**Reply by embraceentropy** [reply]
I'm using the pre-compiled 5x5 manual mesh. Is there a way to revert the BIN file and review the code? Also, I don't see a 5x5 manual template for self-compiling to review. What parameter should I focus on for this issue?

**Reply by xantrk** [reply]
https://github.com/Jyers/Marlin/discussions/597#discussioncomment-705479

**Reply by embraceentropy** [reply]
So, basically, the change on line 601 (in green) should address this issue? I've never self-compiled, so I guess I'll have to try it if so.

https://github.com/MarlinFirmware/Marlin/commit/1efe8a546dc9739597a000c2729e6a379faccd66

**Reply by xantrk** [reply]
> So, basically, the change on line 601 (in green) should address this issue? I've never self-compiled, so I guess I'll have to try it if so.
> 
> [MarlinFirmware@1efe8a5](https://github.com/MarlinFirmware/Marlin/commit/1efe8a546dc9739597a000c2729e6a379faccd66)

I believe so. Without it, it goes to 0 relative position. If it loaded more than it unloaded, it reverts back, or vice versa. Also, using relative or absolute extruder changes the direction of unintended movement as far as I understand. You can self update or wait for @Jyers to pull upstream.

**Reply by embraceentropy** [reply]
I see this line in pause.cpp under Marlin->src->Feature. If I make the change there it will show up when I compile the manual mesh template? Sorry if this is elementary question. I've been wanting to get into compiling my own firmware and figure this is the push I need. I can start a new discussion if needed.

**Reply by Cal-LF** [reply]
I have been trying to make this change but get error right away. Retried and is now compiling.

**Reply by Cal-LF** [reply]
Refer to #253 for steps to compile.

**Reply by Jyers** [reply]
@embraceentropy Great! I'll pull from upstream to address the issue.

**Reply by embraceentropy** [reply]
@Jyers sweet! Is there a way I can try to self-compile with this fix or do I need to wait until you push your next update (based on @Cal-LF error)?

**Reply by Cal-LF** [reply]
I am able to verify this fix to the pause.ccp file works. My filament change worked as it should. 
@embraceentropy you can compile as per the steps in comment #253. I was able to modify the pause.ccp file by adding to line 601 indicated in the info above.
I noted the Jyres pause.ccp file does not have this change in the 3X3 Manual Mesh template. This means the precompiles are using a different source than the templates. I believe this is why the precompiles don't have this issue.

**Reply by embraceentropy** [reply]
@Cal-LF I was on precompiled 5x5 manual and having the issue, though.

**Reply by Cal-LF** [reply]
Well that makes a different story then.

**Reply by embraceentropy** [reply]
Compiled my first firmware and added the pause.cpp fix. All is well now and I learned how to configure and compile my own FW build. All-in-all, a productive day! Thank you!

**Comment by davidhelwi** [comment]
I got the same issue, but it only occurs when I'm trying to change the filament mid print.

**Reply by xantrk** [reply]
https://github.com/Jyers/Marlin/discussions/597#discussioncomment-705479

**Comment by Cal-LF** [comment]
The new merge from Marlin Bugfix has added the fix to the "pause.cpp" file so that no longer has to be user compiled. 
Also see a "leveling fade height" adjustment. No longer have to change it by user compile.
